synapse/rest/key/v2/remote_key_resource.py

```diff
@@ -157,14 +158,22 @@ async def query_keys(
     ) -> JsonDict:
         logger.info("Handling query for keys %r", query)
 
-        store_queries = []
+        cached: Dict[Tuple[str, str], Optional[FetchKeyResultForRemote]] = {}
         for server_name, key_ids in query.items():
-            if not key_ids:
-                key_ids = (None,)
-            for key_id in key_ids:
-                store_queries.append((server_name, key_id))
+            if key_ids:
```

Comment:
Aside: it'd be great if we could give `query` a better annotation

Reply:
I did toy with making a model for it, but thought this PR was big enough as it is